modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackManager.kt

```diff
@@ -960,26 +960,38 @@ open class PlaybackManager @Inject constructor(
                     }
                 }
             }
-
-            playbackStateRelay.blockingFirst().let { playbackState ->
-                val updatedItems = playbackState.chapters.getList().map {
-                    if (it.index == chapter.index) {
-                        it.copy(selected = select)
-                    } else {
-                        it
+            withContext(Dispatchers.Main) {
+                playbackStateRelay.blockingFirst().let { playbackState ->
+                    val updatedItems = playbackState.chapters.getList().map {
+                        if (it.index == chapter.index) {
+                            it.copy(selected = select)
+                        } else {
+                            it
+                        }
                     }
-                }
 
-                playbackStateRelay.accept(
-                    playbackState.copy(
-                        chapters = playbackState.chapters.copy(items = updatedItems),
-                        lastChangeFrom = LastChangeFrom.OnChapterSelectionToggled.value,
-                    ),
-                )
+                    playbackStateRelay.accept(
+                        playbackState.copy(
+                            chapters = playbackState.chapters.copy(items = updatedItems),
+                            lastChangeFrom = LastChangeFrom.OnChapterSelectionToggled.value,
+                        ),
+                    )
+                }
             }
         }
     }
 
+    fun updatePlaybackStateDeselectedChapterIndices() {
+        playbackStateRelay.blockingFirst().let { playbackState ->
```

Comment:
Similarly here. While there is no thread defined here, the usage of this method in the `MainActivity` flow will block the main thread.

https://github.com/Automattic/pocket-casts-android/blob/6c139b2573a64d6c62cb35e3c2fbad65e35a140b/app/src/main/java/au/com/shiftyjelly/pocketcasts/ui/MainActivity.kt#L763

Reply:
I wrapped it in `launch{...}` in https://github.com/Automattic/pocket-casts-android/pull/1916/commits/c75e76036e69db6f167119e11cf3c809801ed2db